packages/components/src/internal/renderers/AttachmentCard.tsx

```diff
@@ -0,0 +1,138 @@
+    }, [setShowModal]);
+
+    const _onDownload = useCallback((): void => {
+        onDownload?.(attachment);
+    }, [attachment, onDownload]);
+
+    const _onRemove = useCallback(() => {
+        if (allowRemove) {
+            onRemove?.(attachment);
+        }
+    }, [allowRemove, attachment, onRemove]);
+
+    if (!attachment) {
+        return null;
+    }
+
+    const { iconFontCls, loadingState, name, size } = attachment;
+    const isLoaded = !isLoading(loadingState);
+    const recentlyCreated = attachment.created ? attachment.created > now() - 30000 : false;
+    const _isImage = isImage(attachment.name);
+
+    return (
+        <>
+            <div className="attachment-card" title={name}>
+                <div
+                    className="attachment-card__body"
+                    onClick={isLoaded ? (_isImage ? _showModal : _onDownload) : undefined}
+                >
+                    <div className="attachment-card__icon">
+                        {_isImage && !isLoaded && <LoadingSpinner msg="" />}
+                        {_isImage && isLoaded && (
+                            <img className={`attachment-card__icon_img ${imageCls}`} src={imageURL} alt={name} />
+                        )}
+                        {!_isImage && <i className={`attachment-card__icon_tile ${iconFontCls}`} />}
+                    </div>
+                    <div className="attachment-card__content">
+                        <div className="attachment-card__name">{name}</div>
+                        <div className="attachment-card__size">
+                            {!isLoaded && <LoadingSpinner msg="Uploading..." />}
+                            {isLoaded && recentlyCreated && (
+                                <>
+                                    <i className="fa fa-check-circle" /> File attached
+                                </>
+                            )}
+                            {isLoaded && !recentlyCreated && size && formatBytes(size)}
+                        </div>
+                    </div>
+                </div>
+                {isLoaded && (
+                    <Dropdown className="attachment-card__menu" componentClass="div" id="attachment-card__menu">
+                        <Dropdown.Toggle useAnchor={true}>
+                            <i className="fa fa-ellipsis-v" />
+                        </Dropdown.Toggle>
+                        <Dropdown.Menu className="pull-right">
+                            <MenuItem onClick={_onDownload}>Download</MenuItem>
+                            {allowRemove && <MenuItem onClick={_onRemove}>Remove attachment</MenuItem>}
+                        </Dropdown.Menu>
+                    </Dropdown>
+                )}
+            </div>
+
+            <Modal bsSize="large" show={showModal} onHide={_hideModal}>
+                <Modal.Header closeButton>
+                    <Modal.Title>
+                        <a onClick={_onDownload} style={{ cursor: 'pointer' }}>
```

Comment:
Perhaps add a title on this to tell people what will happen when they click on the title?

Reply:
done